python/fastquant/strategies/base.py

```diff
@@ -172,261 +190,307 @@ def notify_order(self, order):
+                )
+                self.logging.order_created(
+                    order_type_str="buy",
+                    text="Order type: %s Price: %.2f"
+                    % ("Stop Order", stop_price),
+                )
+
+            # create trailing stop order
+            if self.stop_trail_percent:
+                self.sell(
+                    exectype=bt.Order.StopTrail,
+                    trailpercent=self.stop_trail_percent,
+                    size=final_size,
+                )
+                self.logging.order_created(
+                    order_type_str="buy",
+                    text="Order type: %s Trail Percent: %.2f"
+                    % ("Trailing Stop", self.stop_trail_percent),
+                )
+
+    def create_short_sell_order(self, sell_proportion=None):
+        # Sell short based on the closing price of the previous day
+        if self.execution_type == "close":
+
+            sell_proportion_size = int(
+                self.sell_proportion
+                * self.broker.getvalue()
+                / self.dataclose[1]
+            )
+            # The max incremental short allowed is the short that would lead to a cumulative short position
+            # equal to the maximum short position (initial cash times the maximum short ratio, which is 1.5 by default)
+            max_position_size = max(
+                int(
+                    self.broker.getvalue()
+                    * self.max_short_ratio
+                    / self.dataclose[1]
+                )
+                + self.position.size,
+                0,
+            )
+            if max_position_size > 0:
                 if self.transaction_logging:
                     self.log("SELL CREATE, %.2f" % self.dataclose[1])
-                # Sell a 5% sell position (or whatever is afforded by the current stock holding)
-                # "size" refers to the number of stocks to purchase
-                if self.execution_type == "close":
-                    if SELL_PROP == 1:
-                        self.order = self.sell(
-                            size=self.position.size, exectype=bt.Order.Close
-                        )
-                    else:
-                        # Sell based on the closing price of the previous closing day
-                        self.order = self.sell(
-                            size=int(
-                                (stock_value / (self.dataclose[1])) * self.sell_prop
-                            ),
-                            exectype=bt.Order.Close,
-                        )
-                else:
-                    # Sell based on the opening price of the next closing day (only works "open" data exists in the dataset)
+                self.order = self.sell(
+                    size=min(sell_proportion_size, max_position_size)
+                )
+
+        # Buy based on the opening price of the next closing day (only works "open" data exists in the dataset)
+        else:
```

Comment:
@jbdelmundo I notice that for `create_buy_order` the execution types are `market`, and `close`, while for `create_short_sell_order`, you stuck to the original `close`, and `open` execution types. Is there a reason for why they're not consistent?